openpype/plugins/load/delivery.py

```diff
@@ -0,0 +1,308 @@
+
+        report_items = collections.defaultdict(list)
+
+        selected_repres = self._get_selected_repres()
+        anatomy = Anatomy(self.project)
+        datetime_data = config.get_datetime_data()
+        template_name = self.dropdown.currentText()
+        format_dict = get_format_dict(anatomy, self.root_line_edit.text())
+        for repre in self._representations:
+            if repre["name"] not in selected_repres:
+                continue
+
+            repre_path = path_from_represenation(repre, anatomy)
+
+            anatomy_data = copy.deepcopy(repre["context"])
+            new_report_items = check_destination_path(str(repre["_id"]),
+                                                      anatomy,
+                                                      anatomy_data,
+                                                      datetime_data,
+                                                      template_name)
+
+            report_items.update(new_report_items)
+            if new_report_items:
+                continue
+
+            args = [
+                repre_path,
+                repre,
+                anatomy,
+                template_name,
+                anatomy_data,
+                format_dict,
+                report_items,
+                self.log
+            ]
+
+            if repre.get("files"):
+                for repre_file in repre["files"]:
+                    src_path = anatomy.fill_root(repre_file["path"])
+                    args[0] = src_path
+                    new_report_items, uploaded = process_single_file(*args)
+                    report_items.update(new_report_items)
+                    self._update_progress(uploaded)
+            else:  # fallback for Pype2 and representations without files
+                frame = repre['context'].get('frame')
+                if frame:
+                    repre["context"]["frame"] = len(str(frame)) * "#"
+
+                if not frame:
+                    new_report_items, uploaded = process_single_file(*args)
+                else:
+                    new_report_items, uploaded = process_sequence(*args)
+                report_items.update(new_report_items)
+                self._update_progress(uploaded)
+
+        self.text_area.setText(self._format_report(report(report_items),
+                                                   report_items))
+        self.text_area.show()
+
+        self.resize(self.width(), self.height() + 125)
+
+    def _get_representation_names(self):
+        """Get set of representation names for checkbox filtering."""
+        return set([repre["name"] for repre in self._representations])
+
+    def _get_templates(self, project_name):
+        """Adds list of delivery templates from Anatomy to dropdown."""
+        settings = get_anatomy_settings(project_name)
+        templates = {}
+        for template_name, value in settings["templates"]["delivery"].items():
+            templates[template_name] = value
```

Comment:
Template value that does not start with `{root` is invalid template for delivery and should not be displayed at all.

Reply:
### Side note:
BTW as you should check for `{root` in value you can use `Anatomy` object instead (which is already created in `__init__`) and could maybe store `self.anatomy` instead of `self.project` and then use `anatomy.templates["delivery"].items()`.